src/FormatNumber/Parser.elm

```diff
@@ -378,3 +380,61 @@ parse locale original =
                 | prefix = locale.zeroPrefix
                 , suffix = locale.zeroSuffix
             }
+
+
+{-| Given a `Locale` parses a `String` into a `Maybe Float`
+-}
+parseString : Locale -> String -> Maybe Float
+parseString locale value =
+    let
+        isNegative : Bool
+        isNegative =
+            negativePrefixMatch && negativeSuffixMatch
+
+        negativePrefixMatch : Bool
+        negativePrefixMatch =
+            value
+                |> String.startsWith locale.negativePrefix
+
+        negativeSuffixMatch : Bool
+        negativeSuffixMatch =
+            value
+                |> String.endsWith locale.negativeSuffix
+
+        chrs : Regex.Regex
+        chrs =
+            "[^\\d]"
+                |> Regex.fromString
+                |> Maybe.withDefault Regex.never
+
+        splitValue : String -> List String
+        splitValue nbr =
+            nbr
+                |> String.split locale.decimalSeparator
+                |> List.map (Regex.replace chrs (\_ -> ""))
+
+        sumNumber : Float -> Float -> Float -> Float
+        sumNumber i d pow =
+            if isNegative then
+                -1 * (i + d / (10 ^ pow))
+
+            else
+                i + d / (10 ^ pow)
+    in
+    case splitValue value of
```

Comment:
As we expect to return a `Float`, I would avoid working with a list — a data type that could have none, one, two or more items. I would use a tuple (guaranteed two items: integers, and decimals in our case) instead.

Or even a custom type `{ decimals: Int, integers: Int}`. Let's [make impossible states impossible](https://www.youtube.com/watch?v=IcgmSRJHu_8) — but a tuple `(Float, Float)` works fine here : )

Reply:
Agreed. `joinParts` uses `(String, String)` as input to ensure that only two parts are passed as input.